src/SegmentedMsg.chpl

```diff
@@ -62,6 +63,11 @@ module SegmentedMsg {
           var segArray = getSegArray(segs.a, vals.a, st);
           segArray.fillReturnMap(rtnmap, st);
         }
+        when (DType.BigInt) {
+          var vals = toSymEntry(values, bigint);
+          var segArray = getSegArray(segs.a, vals.a, st);
```

Comment:
This works and is not an issue. I did notice that we could use `assembleSegStringFromParts(segs, vals)` and just use the `SymEntry` instead of creating additional ones. Could we add a separate issue to update each case after this PR is merged since this matches the current format.